src/glue/SessionHL.cpp

```diff
@@ -189,8 +155,8 @@ std::map<std::string, memgraph::communication::bolt::Value> SessionHL::Pull(Sess
                                                                             std::optional<int> n,
                                                                             std::optional<int> qid) {
   try {
-    TypedValueResultStream<TEncoder> stream(encoder, interpreter_context_);
-    return DecodeSummary(interpreter_->Pull(&stream, n, qid));
+    TypedValueResultStream<TEncoder> stream(encoder, interpreter_.db_->storage());
```

Comment:
As said before not guaranteed that all pulled values had the same storage provenance

Reply:
I would leave this as is and have the initial decoupled interpreter work with a single database at a time.